pipeline/transform_gog.py

```diff
@@ -0,0 +1,558 @@
+
+    #Not on discount currently
+    if discount is None:
+        return False
+
+    if not isinstance(discount, str):
+        print("%s is not a valid discount, not a string", discount)
+        return False
+
+    discount = discount.strip().replace('.','')
+
+    if not discount.isnumeric():
+        print("%s is not a valid price, not numeric.", discount)
+        return False
+
+    if not 0 <= int(discount) <= 100:
+        print("%s is not a valid discount, not between 0 and 100.", discount)
+        return False
+
+    return True
+
+
+def is_valid_release(release: str,
+                     days_before_today_allowed=DAYS_BEFORE_TODAY_THAT_WILL_BE_ACCEPTED) -> bool:
+    """Returns true if release is valid."""
+
+    if not isinstance(release, str):
+        print("%s is not a string, not a valid release date.", release)
+        return False
+
+    try:
+        release = release[:10]
+    except:
+        print("%s is not a string, not a long enough to be  date.", release)
+
+
+    try:
+        datetime_release = datetime.strptime(release, "%Y-%m-%d")
+    except Exception as e:
+        print("""%s is not in the valid release form.
+                                %s""", release, e)
+        return False
+
+    earliest_allowed_date = datetime.now().date() - timedelta(days=days_before_today_allowed)
+
+    if not earliest_allowed_date <= datetime_release.date() <= datetime.now().date():
+        print("%s is not within the allowed release date range.", release)
+        return False
+
+    return True
+
+
+def is_valid_image(image: str) -> bool:
+    """Returns true if image is valid."""
+
+    if not isinstance(image, str):
+        print("%s is not a valid image, not a string", image)
+        return False
+
+    image = image.replace(" ",'')
+    image = image.replace("\n",'')
+    image = image.split(',')[0]
+    image = image.strip()
+
+    if len(image) > 256:
+        print("%s is not a valid image, url too long", image)
+        return False
+
+    if len(image) == 0:
+        print("%s is not a valid image, empty string", image)
+        return False
+
+    try:
+        response = get(image, timeout=5)
+    except Exception as e:
+        print("""%s is not a valid image, not loading properly.
+                        Error: %s""", image, e)
+        return False
+
+    if not response.status_code == 200:
+        print("%s is not a valid image, not loading properly.", image)
+        return False
+
+    return True
+
+
+def is_valid_age(age: str) -> bool:
+    """Returns if the age conforms to PEGI standards."""
+
+    if age is None:
+        return False
+
+    if not isinstance(age, str):
+        print("%s is not a valid age rating, not a string", age)
+        return False
+
+    age = age.strip()
+
+    if age not in ['3', '7', '12', '16', '18']:
+        print("%s is not a valid age rating, not a standard PEGI age", age)
+        return False
+
+    return True
+
+
+def format_data(game: dict) -> bool:
```

Comment:
This is kinda hard to read. Instead of repetitive if, else statements you could:
```suggestion
def format_data(game: dict) -> bool:
    fields = {
        "publisher": (is_valid_publisher, format_publisher_list, []),
        "developer": (is_valid_developer, format_developer_list, []),
        "tag": (is_valid_tag, format_tag_list, []),
  }

    formatted_data = {}

    for field, (validator, formatter, default) in fields.items():
        formatted_data[field] = formatter(game[field]) if validator(game[field]) else default

    return formatted_data
```

Reply:
Obviously add all of the parts requiring formatting.